docs/advanced/api_examples.yml

```diff
@@ -209,6 +209,30 @@
 
 -
   id:
+    "update dataset size"
+
+  doc: |
+    Update a dataset with a new maximum_size. This will grow or shrink
+    the dataset.
```

Comment:
Should this also say something about filesystems?  I'm not sure.  We aren't super precise in our use of "volumes" vs "datasets" and we hardly ever talk about "filesystems".  Yet I'm pretty I would, as a user of this API, want to know what the state of my filesystem is after a resize operation like this.